plugins/discovery-ec2/build.gradle

```diff
@@ -102,6 +102,7 @@ tasks.named("test").configure {
   } else {
     nonInputProperties.systemProperty 'java.security.policy', "file://${buildDir}/tmp/java.policy"
   }
+  environment 'AWS_METADATA_SERVICE_TIMEOUT', '1'
```

Comment:
why does that need to passes in here? can't we hardcode it in the test?

Reply:
We can't set environment variables in a unit test, so we have to configure it before running them

Reply:
The test should work without explicitly setting the environment variable? It would just take a bit longer, 2 seconds vs 1 second. If that's the case, I think it is worth to randomize it, e.g.
```
if (BuildParams.random.nextBoolean()) {
  env 'AWS_METADATA_SERVICE_TIMEOUT', '1'
}
```